fix: Scalar parameter docstring and example arguments unused

maxburs opened the pull request:
- Fixed scalar parameter docstring and example arguments not passed along when creating ctor versions
- Improved Playwright config
  - Removed our custom timeout config, which was much lower than the defaults
  - Use recommended reporters in CI (github + html)
  - Upload html report, so we can see why tests failed

maxburs commented on package/src/languageServiceManager/kustoLanguageService.ts:
```diff
@@ -1695,7 +1695,7 @@ class KustoLanguageService implements LanguageService {
             getCslTypeNameFromClrType(param.type)
         );
-        return new sym.ParameterSymbol(param.name, paramSymbol, null);
+        return new sym.ParameterSymbol(param.name, paramSymbol, param.docstring ?? null);
```
Tested this change, others are a bit more speculative

morgilad replied:
I'm not clear on the use case. 
Could we add a unit or integration test to clarify it and help prevent regressions?

maxburs replied:
Examples and docstring/description are added to tooltips on hover. I'm using them for dashboard system parameters, and and I'd assume other ones are in the system we get from kusto (otherwise, IDK why the C# api would expose it)

> Could we add a unit or integration test to clarify it and help prevent regressions?

Unlike a lot of other stuff, this isn't really load bearing. If it breaks it won't block a release. I don't think it should be a priority for testing.

maxburs replied:
If a unit test would help I might add that, because it would be easy, but I don't think it would do anything useful.

An integration tests would be pretty expensive to add.

morgilad replied:
I'm sorry if I wasn't clear. 
From my perspective, testing is an important part of our work as engineers. 
It helps ensure that what we build remains reliable and understandable for other developers. 
I would suggest adding an integration test here, but if that's too difficult, a unit test would also be fine.

sagivf replied:
@maxburs can you add some examples/screenshots of there this shows up and how?
That would help in understanding how/what test we should add.

maxburs replied:
<img width="618" alt="image" src="https://github.com/user-attachments/assets/63213c75-95d3-42bd-9030-5dccd6eaae1f">

sagivf commented:
@maxburs I'd like this to be tested. This test will easily find other related regressions.

I've opened a task this sprint to complete this task (with an integration test :) )
https://msazure.visualstudio.com/One/_workitems/edit/29323167

maxburs commented:
@morgilad Tests added

morgilad commented:
@maxburs the build is failing because of issues with other tests.

maxburs commented:
@morgilad CI is passing 🎊

Changes since you last took a look:
- Removed Playwright timeout config. This was much lower than the defaults; removing it fixed the CI failures.
- Updated the config to use the github + html reporters in CI
- Moved the "retain-on-failure" trace config from the cli to the config object
- Configured CI to upload the html Playwright report

maxburs commented on package/tests/integration/playwright.config.ts:
```diff
@@ -1,16 +1,16 @@
     outputDir: './.test-results',
     preserveOutput: 'failures-only',
     forbidOnly: !!process.env.CI,
-    retries: process.env.CI ? 2 : 0,
```
This is the default, configuring it is not necessary

morgilad replied:
I prefer to be explicit actually.

maxburs commented on package/tests/integration/playwright.config.ts:
```diff
@@ -1,16 +1,16 @@
     workers: process.env.CI ? 1 : undefined,
-    timeout: 5_000,
-    expect: { timeout: 5_000 },
-    reporter: 'line',
```
Line is the default outside of CI, so configuring it is not necessary

morgilad replied:
Thank you

morgilad commented on .github/workflows/pr.yml:
```diff
@@ -41,3 +41,11 @@ jobs:
         run: yarn test:integration
+
+      # https://playwright.dev/docs/ci-intro#html-report
+      - uses: actions/upload-artifact@v4
```
Where can I see this report?
